src/steps/relay.py

```diff
@@ -34,6 +45,25 @@ def setup_main_relay_nodes(self, request):
         self.node2.start(relay="true", discv5_bootstrap_node=self.enr_uri)
         self.main_nodes.extend([self.node1, self.node2])
 
+    @pytest.fixture(scope="function")
+    def setup_main_rln_relay_nodes(self, request):
+        logger.debug(f"Running fixture setup: {inspect.currentframe().f_code.co_name}")
+        rln_creds = {
+            "eth_client_address": ETH_CLIENT_ADDRESS,
+            "go_waku_eth_client_private_key": GO_WAKU_ETH_TESTNET_KEY,
+            "nwaku_eth_client_private_key": NWAKU_ETH_TESTNET_KEY,
+            "keystore_password": KEYSTORE_PASSWORD,
+            "eth_contract_address": ETH_CONTRACT_ADDRESS,
+        }
+        self.node1 = WakuNode(NODE_1, f"node1_{request.cls.test_id}")
+        self.node1.start(relay="true", rln_creds=rln_creds, rln_register_only=True)
+        # self.node1.start(relay="true", nodekey=NODEKEY, rln_creds=rln_creds)
+        # self.enr_uri = self.node1.get_enr_uri()
+        self.node2 = WakuNode(NODE_2, f"node2_{request.cls.test_id}")
+        self.node2.start(relay="true", rln_creds=rln_creds, rln_register_only=True)
```

Comment:
It is trickier a bit here. We should ideally use different private key for different node. Not sure how could smart contract handle situation where two nodes modify its state under same account address. I was assuming only two nodes Go/Nim Waku for starter. If we need more, I would need to change code to accept node numbers and generate private keys and fund accounts automatically.

Reply:
hmm but from what I see in this PR we use the same keys for all nodes, and /keystore/keystore.json will use the data of the last registered node
TBH I prefer to keep it simple at this point, and reuse the registration if possible, That way we can spin additional nodes much more easily

Reply:
We may try to keep it simple for this PR, with a risk tests reusing the same keys may fail and we would need to include inference in smart contract state as possible cause.

Reply:
If the tests fail (because sharing the keys is not allowed) then indeed we will need at least 2 set of keys.
But otherwise, if it works, I would take the simple path for now. 
Not sure I got your meaning regarding the smart contract

Reply:
I was checking with developers. They rise question "But then I imagine it's possible you'd end up sending more than one message per epoch". Node may actually spam on behalf of another node.

Reply:
Ok, I see, thanks for checking!
But I have some questions:

1. We need a unique set of `eth-client-address`, `eth-private-key` and `cred-password` for each node we run? I see in the env file now 1 ETH_CLIENT_ADDRESS, 3 ETH_TESTNET_KEY (1 simple, one for nwaku and onde for gowaku) and 1 KEYSTORE_PASSWORD and not sure why is that
2. If we are going to have multiple keys, does it make sense to store them in a a better data structure like a JSON (instead of ETH_CLIENT_ADDRESS, ETH_TESTNET_KEY  ) that can be easily extended with more keys for more nodes?
3. Will each registration append the node data to `/keystore/keystore.json` or overwrite this file? If it overwrites don't we need unique keystores for each registration?

Reply:
1. We need unique private keys (Ethereum accounts). Storage path inside docker could be same for each node. Outside filesystem directory needs unique name. I have used prefix like /go-waku, and I can change it to numbered directories for example.
2. Yes, we can have entire JSON stored in Github secrets https://stackoverflow.com/questions/72177570/how-to-pass-github-secrets-as-value-in-json-file
3. Yes, we would need unique directory names for outside docker (OS).

Reply:
sounds good, thanks